pandas/tests/indexes/datetimes/test_datetime.py

```diff
@@ -457,3 +457,18 @@ def test_to_frame_datetime_tz(self):
         result = idx.to_frame()
         expected = DataFrame(idx, index=idx)
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize("name", [None, "name"])
+    def test_index_map(self, name):
+        # see GH20990
+        count = 6
+        index = pd.date_range("2018-01-01", periods=count, freq="M", name=name)
+        a = pd.Series(np.arange(count), index=index)
```

Comment:
Can you delete the Series altogether and just use `tm.assert_index_equal`? I don't think the Series is required at all

Reply:
Do you mean to make it look like following?

```
@pytest.mark.parametrize("name", [None, "name"]) 
def test_index_map(self, name):
        # see GH20990
        count = 6
        index = pd.date_range("2018-01-01", periods=count, freq="M", name=name).map(lambda x: (x.year, x.month))
        expected = pd.MultiIndex.from_product(
            ((2018,), range(1, 7)), names=[name, name]
        )
        tm.assert_index_equal(index, expected)
```

Reply:
@WillAyd ping